### ISSUE TITLE
Addressing `DeprecationWarning` Notices in `clearml`

### ISSUE THREAD

**Issue by AH-Merii** [allegroai/clearml#1145]
Hi all,

I wasn't sure what to submit this under, I'm pretty sure it's not a feature request, however, it's not a bug either, so apologies in advance for the confusing Label 😅.

While working with the library in a Python 3.10 environment, I've come across a few `DeprecationWarning` notices. I wanted to bring them to your attention and also suggest potential fixes.

1. **jsonschema FormatError Import**:

   **Warning**:
   ```
   Importing FormatError directly from the jsonschema package is deprecated and will become an ImportError. Import it from jsonschema.exceptions instead.
   ```
   **Suggested Fix**:
   Update the import in `clearml/backend_api/session/request.py` from:
   ```python
   from jsonschema import FormatError
   ```
   to:
   ```python
   from jsonschema.exceptions import FormatError
   ```

2. **jsonschema RefResolutionError Deprecation**:

   **Warning**:
   ```
   jsonschema.exceptions.RefResolutionError is deprecated as of version 4.18.0. If you wish to catch potential reference resolution errors, directly catch referencing.exceptions.Unresolvable.
   ```
   **Suggested Fix**:
   Update the import in `clearml/backend_api/session/request.py` from:
   ```python
   from jsonschema.exceptions import RefResolutionError
   ```
   to:
   ```python
   from jsonschema.referencing import Unresolvable
   ```

3. **Accessing attr.__version__**:

   **Warning**:
   ```
   Accessing attr.__version__ is deprecated and will be removed in a future release. Use importlib.metadata directly to query for attrs's packaging metadata.
   ```
   **Suggested Fix**:
   In `clearml/utilities/attrs.py`, instead of accessing `attr.__version__` directly, use:
   ```python
    from importlib.metadata import version
    
    ...
    
    if Version(version("attrs")) >= Version("19.2"):
        cmp = kwargs.pop("cmp", None)
        if cmp is not None:
            kwargs["eq"] = kwargs["order"] = cmp

   ```

**Why Address These Warnings?**
While deprecation warnings aren't errors, they can be indicators of changes that will occur in future releases of the libraries `clearml` depends on. Addressing them ensures that `clearml` remains compatible with these libraries as they evolve. Plus, it provides a cleaner experience for users who won't see these warnings.

I hope these suggestions help! Thanks again for all your hard work on this project. 

Looking forward to future versions of `clearml`.

## Environment
* Server type: **self-hosted**
* ClearML SDK Version: `1.13.1`
* ClearML Server Version: `1.12.1`
* Python Version: `3.10.13`
* OS: **WSL2**

**Comment by jkhenning** [comment]
@AH-Merii these were already addressed in https://github.com/allegroai/clearml/pull/1128 and will probably be merged soon after the PR is ready

**Comment by AH-Merii** [comment]
@jkhenning I believe only the jsonschema issues where discussed (point 1 & 2) in #1128 and not point 3.

**Comment by jkhenning** [comment]
@AH-Merii quite right, I'll fix the third issue 🙂

**Comment by pollfly** [comment]
Hey @AH-Merii ! Just letting you know that this issue has been resolved in the recently released [v1.13.2](https://github.com/allegroai/clearml/releases/tag/v1.13.2). Let us know if there are any issues :)
